packages/nextjs/components/modals/CreateDealModal.tsx

```diff
@@ -57,18 +61,106 @@ export function CreateDealModal({ onSuccess, deal, setDeal }: CreateDealModalPro
   const updateRequirements = (newRequirements: string) => {
     setDeal({ ...deal, requirements: newRequirements });
   };
+  const { address, setDealId, setSymmetricKey } = useGlobalState();
 
-  // ADD: Deal creation logic
-  const createDeal = () => {
-    const isDealCreated = true;
+  const createDeal = async () => {
+    const paymentPerThousandInApeWei =
+      (BigInt(Math.round(deal.paymentPerThousand * 10000)) * BigInt(10 ** 18)) / BigInt(10000);
 
-    notification.info("ADD: Deal creation logic");
+    const privateTerms = {
+      twitterUserId: deal.twitterHandle,
+      paymentPerLike: (paymentPerThousandInApeWei / BigInt(1000)).toString(),
+    };
 
-    // Note: Keep these at the end
-    if (isDealCreated) {
-      setOpen(false); // Closes 'Create Deal' modal
-      onSuccess(); // Opens 'Deal Sent' modal
-    }
+    const symKey = await crypto.subtle.generateKey(
+      {
+        name: "AES-CBC",
+        length: 256,
+      },
+      true,
+      ["encrypt", "decrypt"],
+    );
+    const exportedSymKey = await crypto.subtle.exportKey("raw", symKey);
+    const exportedSymKeyBase64 = btoa(String.fromCodePoint(...new Uint8Array(exportedSymKey)));
+    setSymmetricKey(exportedSymKeyBase64);
```

Comment:
I really hope that we could get Lit to work, if we have the time of course, this really scratches me the wrong way 😅